src/server/database/database_provider.py

```diff
@@ -0,0 +1,113 @@
+import os
+import sqlite3
+from contextlib import contextmanager
+from typing import Generator, Optional
+
+from .. import PATHS
+from . import DatabaseResponse, Message
+
+
+class DatabaseProvider:
+    """Interface to SQL database.
+
+    Example usage:
+        DatabaseProvider.initialize()  # init database
+        age: int = 66
+        with DatabaseProvider.query(
+            "SELECT name FROM person WHERE age=?", (age,)
+        ) as response:
+            if response.message is Message.OK:
+                name: list[str] = response.data
+            else:
+                # handle query failure
+    """
+
+    connection: sqlite3.Connection = sqlite3.Connection(":memory:")
+    database_source: str = ""
+
+    @classmethod
+    def initialize(cls) -> None:
+        """Initialize connection to the database."""
+        cls.database_source = PATHS.DATABASE
+
+        if not os.path.exists(cls.database_source):
+            result = cls._create_database()
+            if result is Message.NO_CONNECTION:
+                print("INFO: connecting to in-memory database as fallback")
+                cls.database_source = ":memory:"
+                cls._create_database()
+        cls._fill_database()
```

Comment:
To nie wywala błędu tylko dlatego, że bazę danych masz w pamięci. Powinno to być wykonane tylko jeśli baza jeszcze nie istnieje. Po zamianie w linijce 25 ścieżki na PATHS.DATABASE wywala błędy. Po wstawieniu tego do warunku, przestaje wywalać i dane zapisują się w bazie danych.

Reply:
Fixed.